mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java

```diff
@@ -263,6 +263,29 @@ public void searchingMessageInMultipleMailboxShouldNotReturnLessMessageThanLimit
             .hasSize(10);
     }
 
+    @Test
```

Comment:
I added this test because your test (above), did not fail with your previous code:
```
       return ImmutableList.copyOf(FluentIterable.from(temp)
                .transform(toMessageId())
                .limit(Long.valueOf(limit).intValue())
                .toSet());
```
whereas this one does.